labs/iot/gps-lab/client/code/raspberry-pi-gps-tracker.py

```diff
@@ -0,0 +1,95 @@
+import time
+import serial
+
+# Using the Python Device SDK for IoT Hub:
+#   https://github.com/Azure/azure-iot-sdk-python
+# The sample connects to a device-specific MQTT endpoint on your IoT Hub.
+from azure.iot.device import IoTHubDeviceClient, Message
+
+# The device connection string to authenticate the device with your IoT hub.
+CONNECTION_STRING = "<Device_Connection_String>"
+
+# Define the JSON message to send to IoT Hub.
+MSG_TXT = '{{"latitude": {latitude},"longitude": {longitude}}}'
+
+SERIAL_PORT = "/dev/serial0"
+
+gps = serial.Serial(SERIAL_PORT, baudrate = 9600, timeout = 1.5)
+
+#This method converts a transmitted string from DDMM.MMMMM to DD.MMMM
+def formatDegreesMinutes(coordinates, digits):
+    
+    parts = coordinates.split(".")
+
+    if (len(parts) != 2):
+        return coordinates
+
+    if (digits > 3 or digits < 2):
+        return coordinates
+    
+    left = parts[0]
+    right = parts[1]
+    degrees = str(left[:digits])
+    minutes = str(right[:3])
+
+    return degrees + "." + minutes
+
+# This method reads the data from the serial port, the GPS sensor is attached to,
+# and then parses the NMEA messages it transmits.
+# gps is the serial port, that's used to communicate with the GPS adapter
+def getPositionData(gps):
+    data = gps.readline()
+    message = data[0:6]
+    print(message);
+    if (message.decode('utf-8') == "$GPRMC"):        
```

Comment:
Are there any Python libraries to help with formatting the messages from a GPS device?